### DISCUSSION TITLE
Intercept Network Requests / Responses

### DISCUSSION THREAD

**Discussion by dbrody** [keathley/wallaby#643] [Q&A]
Is there a way to intercept network traffic on page loads / navigation inside Wallaby?
This is a [feature playwright has](https://playwright.dev/docs/network/#network-events) and would be extremely useful in the elixir/Wallaby land.

For example:
I am navigating to a page and I want to make sure the page calls an API endpoint and I want to inspect / verify the response of that call. However, the data is no within the HTML / content of the page at all.

**Comment by mhanberg** [comment]
Hi,

I don't believe this is possible currently.

Playwright and Puppeteer use the Chrome DevTools Protocol to control the browser, which is a bit different than the way webdriver works.

I have plans to create a Wallaby Driver that uses the CDP, which could enable this functionality, but it would probably come from the underlying library that Wallaby will use (or write), and not Wallaby itself.
